### PR TITLE
Fixes Initial Sync

### PR DESCRIPTION
**Pull request by nisdas**
This follows on from #2050

Currently initial sync doesn't work because whenever it requests a batch of blocks and during the processing of those blocks we have an epoch transition the node will panic. This is due to there being no justified block being saved. 

Therefore as a solution to this, we are having specific state requests and responses for both the finalized and justified states, so as to save them. We also check a boolean on whether to run ffg checkpoints or not. Running the checkpoints would not make sense if we aren't up to date. This should fix any outstanding issues with initial sync.

Major Changes: 

- [x] Save finalized and justified states in inital sync
- [x] Better logging in initial sync(display slots from genesis)

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Please add PR description that explains the problem

File: beacon-chain/blockchain/fork_choice.go [hunk 1]
```diff
@@ -105,24 +115,50 @@ func (c *ChainService) updateFFGCheckPts(state *pb.BeaconState) error {
 
 // ApplyForkChoiceRule determines the current beacon chain head using LMD GHOST as a block-vote
 // weighted function to select a canonical head in Ethereum Serenity.
-func (c *ChainService) ApplyForkChoiceRule(ctx context.Context, block *pb.BeaconBlock, computedState *pb.BeaconState) error {
+func (c *ChainService) ApplyForkChoiceRule(ctx context.Context, block *pb.BeaconBlock, postState *pb.BeaconState) error {
 	ctx, span := trace.StartSpan(ctx, "beacon-chain.blockchain.ApplyForkChoiceRule")
 	defer span.End()
-	h, err := hashutil.HashBeaconBlock(block)
+	log.Info("Applying LMD-GHOST Fork Choice Rule")
```
Comment (IvanTheGreatDev) [inline beacon-chain/blockchain/fork_choice.go]: Should this be a "info" log? How often would it print out?

Comment (nisdas) [reply beacon-chain/blockchain/fork_choice.go]: Everytime fork choice is run, this is a part of #2050. Most of the changes are minor here, the diff for comparison https://github.com/prysmaticlabs/prysm/compare/testruntime...fixSync

File: beacon-chain/sync/initial-sync/service.go [hunk 2]
```diff
@@ -420,6 +420,32 @@ func (s *InitialSync) processState(msg p2p.Message) {
 
 	if err := s.db.SaveCurrentAndFinalizedState(beaconState); err != nil {
 		log.Errorf("Unable to set beacon state for initial sync %v", err)
```
Comment (terenc3t) [inline beacon-chain/sync/initial-sync/service.go]:
```suggestion
		log.Errorf("Could not set beacon state for initial sync %v", err)
```

File: beacon-chain/sync/initial-sync/service.go [hunk 3]
```diff
@@ -420,6 +420,32 @@ func (s *InitialSync) processState(msg p2p.Message) {
...
+		return
+	}
+
+	if err := s.db.UpdateChainHead(beaconState.LatestBlock, beaconState); err != nil {
+		log.Errorf("Could not update chainhead %v", err)
+		return
+	}
+
+	if err := s.db.SaveJustifiedState(beaconState); err != nil {
+		log.Errorf("Unable to set beacon state for initial sync %v", err)
```
Comment (terenc3t) [inline beacon-chain/sync/initial-sync/service.go]:
```suggestion
		log.Errorf("Could not set beacon state for initial sync %v", err)
```

**Pull request merged**